app/models/reminder.rb

```diff
@@ -0,0 +1,22 @@
+class Reminder < ApplicationRecord
+  belongs_to :contract
+
+  validates :remind_at, presence: true
+  validate :remind_at_within_contract_dates
+
+  after_create :schedule_reminder
+
+  private
+
+  def schedule_reminder
+    ReminderJob.set(wait_until: remind_at).perform_later(self)
+  end
+
+  def remind_at_within_contract_dates
+    return if contract.blank? || remind_at.blank?
```

Comment:
If you add a `validates :contract, presence: true` you can remove this condition :)

A Reminder without a contract or a remind_at date seems incorrect, so we shouldn't even allow it to be created, and therefore we do not need to check if these values exist.